mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/ThreadIdGuessingAlgorithm.java

```diff
@@ -0,0 +1,27 @@
+/******************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one     *
+ * or more contributor license agreements.  See the NOTICE file   *
+ * distributed with this work for additional information          *
+ * regarding copyright ownership.  The ASF licenses this file     *
+ * to you under the Apache License, Version 2.0 (the              *
+ * "License"); you may not use this file except in compliance     *
+ * with the License.  You may obtain a copy of the License at     *
+ *                                                                *
+ * http://www.apache.org/licenses/LICENSE-2.0                     *
+ *                                                                *
+ * Unless required by applicable law or agreed to in writing,     *
+ * software distributed under the License is distributed on an    *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY         *
+ * KIND, either express or implied.  See the License for the      *
+ * specific language governing permissions and limitations        *
+ * under the License.                                             *
+ ******************************************************************/
+
+package org.apache.james.mailbox.store.mail;
+
+import org.apache.james.mailbox.model.ThreadId;
+import org.apache.james.mailbox.store.mail.model.MailboxMessage;
+
+public interface ThreadIdGuessingAlgorithm {
+    ThreadId getThreadId(MailboxMessage mailboxMessage);
```

Comment:
```suggestion
    ThreadId getThreadId(MessageId messageId, List<References> references, Subject subject);
```

With

```
public class Reference {
    private final String value;
    
    // ...
}
```

and 

```
public class Subject {
    private final String value;
    
    // ...
}
```

Where references aggegates the In-Reply-To // References header fields.

Reply:
Ok it looks like I understand wrong about naive implementation level. I will working on this.